### PR TITLE
[PLAT-354] Create a test for each needed vesting schedule

### PR DESCRIPTION
**Pull request by DidacSF**
## Description

See https://ajunanetwork.atlassian.net/browse/PLAT-354 for details. In short:

* Add `ajuna-test` crate
* Add vesting schedule tests into `ajuna-test` crate

## Type of changes

- [ ] `build`: Changes that affect the build system or external dependencies (eg, Cargo, Docker)
- [ ] `ci`: Changes to CI configuration
- [ ] `docs`: Changes to documentation only
- [ ] `feat`: Changes to add a new feature
- [ ] `fix`: Changes to fix a bug
- [ ] `refactor`: Changes that do not alter functionality
- [ ] `style`: Changes to format the code
- [x] `test`: Changes to add missing tests or correct existing tests

## Checklist

- [x] Tests for the changes have been added
- [x] Necessary documentation is added (if appropriate)
- [x] Formatted with `cargo fmt --all`
- [x] Linted with `cargo clippy --all-features --all-targets`
- [x] Tested with `cargo test --workspace --all-features --all-targets`

### CODE REVIEW

File: ajuna-test/src/sidechain.rs [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+	type AccountStore = System;
+	type WeightInfo = ();
+}
+
+parameter_types! {
+	pub const MinVestedTransfer: Balance = 100 * MICRO_AJUNS;
+	pub const MaxVestingSchedules: u32 = 100;
+}
+
+impl orml_vesting::Config for SideChainRuntime {
```
Comment (andyjsbell) [inline ajuna-test/src/sidechain.rs:67]: I don't think we need vesting in the sidechain, just in the parent chain I think

File: ajuna-test/src/sidechain.rs [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+				balances: vec![
+					(alice(), 10_000 * EXISTENTIAL_DEPOSIT),
+					(bob(), 20_000 * EXISTENTIAL_DEPOSIT),
+					(charlie(), 30_000 * EXISTENTIAL_DEPOSIT),
+					(dave(), 40_000 * EXISTENTIAL_DEPOSIT),
+					(eve(), 10_000 * EXISTENTIAL_DEPOSIT),
+					(ferdie(), 9_999_000 * EXISTENTIAL_DEPOSIT),
+				],
+			},
+			vesting: orml_vesting::GenesisConfig { vesting: vec![] },
```
Comment (andyjsbell) [inline ajuna-test/src/sidechain.rs:269]: As above, vesting should be on layer 1

File: ajuna-test/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+	use ajuna_solo_runtime::currency::*;
+	use orml_vesting::{VestingSchedule, VESTING_LOCK_ID};
+	use pallet_balances::{BalanceLock, Reasons};
+
+	use crate::{keyring::*, sidechain::*};
+
+	#[test]
+	fn validate_simple_vesting_schedule() {
+		SideChain::<SideChainSigningKey>::build().execute_with(|| {
```
Comment (andyjsbell) [inline ajuna-test/src/lib.rs:105]: I think this would need to be the ajuna-node

File: ajuna-test/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+	/// * The vesting begins at block 10
+	/// * There's an initial cliff period of 100 blocks
+	/// * After the cliff 25% of the total issuance gets unlocked (100 µAJUNS)
+	/// * Past that point the remaining issuance gets linearly unlocked during the following 2000
+	///   blocks, which means that each block unlock 450 nAJUNS
+	///
+	/// In order to perform such scheme we need to define two chained VestingSchedule blocks
+	#[test]
+	fn validate_complex_vesting_schedule() {
+		SideChain::<SideChainSigningKey>::build().execute_with(|| {
```
Comment (andyjsbell) [inline ajuna-test/src/lib.rs:143]: Again here, I guess you know where I am going with this 😄

**Review by andyjsbell — COMMENTED** [review]
Looks good in terms of integrated the pieces so well done.  We need to have the vesting tested on the layer 1 runtime and I think you have put in the layer 2, no problem.  The layer 2 isn't technically needed here as I commented in the ticket so it could be removed if that makes things easier to follow.  I don't mind either way 👍

File: ajuna-test/src/sidechain.rs (hunk 1 above)
Comment (DidacSF) [reply ajuna-test/src/sidechain.rs]: You are right, I've moved the tests to the Node.

**Comment by DidacSF** [comment]
@andyjsbell I've moved all the tests to work with the Node instead, so if you can recheck that would be helpfull!

**Review by andyjsbell — CHANGES_REQUESTED** [review]
LGTM, thanks @DidacSF 👍

**Review by andyjsbell — APPROVED** [review]

**Pull request merged**